### ISSUE TITLE
Matching multiple license headers?

### ISSUE THREAD

**Issue by JoshLind** [Lucas-C/pre-commit-hooks#68]
Hi folks,

Firstly, thanks for building and maintaining this useful tool -- it's great! 😄 One question I have, is there support for matching multiple license files? i.e., check if the license in a file is one of several options? We have use cases where different files require slightly different licenses.

I see that there's support for fuzzy matching (https://github.com/Lucas-C/pre-commit-hooks#fuzzy-license-matching), but it would be great to see if we could support *exact* matching with multiple potential licenses? Does this already exist, or is it something that would require changes? (Apologies if I've missed it!)

Thanks for your help!

**Comment by Lucas-C** [comment]
Hi @JoshLind!

It is actually very simple to have different rules for different files usingn this hook: simply duplicate configuration blocks!
```yaml
  - repo: https://github.com/Lucas-C/pre-commit-hooks
    rev: v1.4.2
    hooks:
      - id: insert-license
        files: \.groovy$
        args:
          - --license-filepath
          - LICENSE
      - id: insert-license
        files: \.java$
        args:
          - --license-filepath
          - LICENSE2
          - --use-current-year
```

Does that answer your use case?

**Comment by JoshLind** [comment]
Hi @Lucas-C!

Thanks for the quick response 😄 If I understand correctly, your example would provide different licenses to files based on type, right? i.e., license 1 for `groovy` files and license 2 for `java` files? Our use-case is different -- all the files will have the same types (and even directory locations), it's just that some will have a different license depending on who helped contribute to and/or write the file.

From a UI perspective, it'd probably look more like:
```
  - repo: https://github.com/Lucas-C/pre-commit-hooks
    rev: v1.4.2
    hooks:
      - id: insert-license
        files: \.rs$ # Rust files
        args:
          - --license-filepath
          - LICENSE1 # The file could have this license
          - --license-filepath
          - LICENSE2 # Or the file could have this license
```

Let me know if what I'm saying doesn't make sense 😃

**Comment by Lucas-C** [comment]
OK, I understand your need better.
Can't you use dfferent `files` selectors in different blocks to select which license content should be used in which files?
Could you maybe provide some examples of the kind of "rules" you plan to have?

**Comment by JoshLind** [comment]
Thanks @Lucas-C 😄 

> Can't you use dfferent files selectors in different blocks to select which license content should be used in which files?
> Could you maybe provide some examples of the kind of "rules" you plan to have?

Unfortunately, not. We have 1k+ files, and given that every single file will have one of two different licenses (depending on which Github accounts contributed to the file), we'd likely need to specify every single file and license individually, which will be a nightmare.

For context, we have a project (Aptos: https://github.com/aptos-labs/aptos-core), which was born out of Meta's Libra project. As a result, we want to give files a different license, based on the files that: (i) contain only our code; vs (ii) contain code from both projects. Does that make more sense? 😄

**Comment by frazar** [comment]
Hi! If you agree, I could draft a PR to implement the described functionality.

My proposal would be to change the linter to accept more than one `license_file` argument. Then, the linter would pass successfully for repo files whose license string matches at least one of the license files passed as argument. 

This would be backward compatible with existing linting rules having a single license file argument.

What do you think?

**Comment by JoshLind** [comment]
Yeah, that would be perfect for our use-case 😄 But, I'm not sure what @Lucas-C thinks.

**Comment by Lucas-C** [comment]
This has been implemented in #88 by @frazar and is now released!
